_posts/2021-11-23-how-to-dry-your-ref-arch.adoc

```diff
@@ -0,0 +1,594 @@
+configurations. However, the Service Catalog based version utilizes constructs that are more friendly to DRY across
+environments. By switching to the Service Catalog based version, you will see more common configuration across your
+environments, allowing you to DRY your configuration further.
+
+An additional consideration is that the Service Catalog upgrade guide assumes the original Reference Architecture, and
+not the DRY version. If you update to the DRY version first, then the guide will no longer be compatible with your
+Reference Architecture, complicating a later migration to the Service Catalog. Updating to the Service Catalog from a
+DRY Reference Architecture version is not fully supported by Gruntwork.
+
+
+=== Prerequisite: Update Terragrunt to a compatible version
+
+This guide assumes you are using Terragrunt version
+https://github.com/gruntwork-io/terragrunt/releases/tag/v0.35.0[v0.35.0] or newer, so that all bug fixes for multiple
+include blocks are patched. Please upgrade to a compatible version prior to moving on in the guide.
+
+
+=== Refactoring common configurations for a component
+
+The rest of this guide will cover how to identify and extract the common configurations for a single arbitrary component
+in your Reference Architecture. These steps can be used to refactor any component that is deployed in multiple
+accounts or environments in your Reference Architecture.
+
+* <<step_1>>
+* <<step_2>>
+* <<step_3>>
+* <<step_4>>
+* <<further_steps>>
+
+[[step_1]]
+==== Step 1: Identify the component
+
+The first step is to decide which component you would like to refactor. A component in this context is a single
+Terraform module that is deployed by Terragrunt.
+
+Once you identify the component that you want to update, take an inventory of the list of Terragrunt configurations that
+deploy that component across your environments. We will use this inventory to identify the common configuration across
+all the deployments.
+
+Example:
+
+* `vpc`
+** `dev/us-west-2/dev/networking/vpc`
+** `stage/us-west-2/stage/networking/vpc`
+** `prod/us-west-2/prod/networking/vpc`
+* `openvpn-server`
+** `dev/us-west-2/dev/networking/openvpn-server`
+** `stage/us-west-2/stage/networking/openvpn-server`
+** `prod/us-west-2/prod/networking/openvpn-server`
+* `ecs-deploy-runner`
+** `dev/us-west-2/mgmt/ecs-deploy-runner`
+** `stage/us-west-2/mgmt/ecs-deploy-runner`
+** `prod/us-west-2/mgmt/ecs-deploy-runner`
+** `shared/us-west-2/mgmt/ecs-deploy-runner`
+** `logs/us-west-2/mgmt/ecs-deploy-runner`
+** `security/us-west-2/mgmt/ecs-deploy-runner`
+
+The changes will be isolated to Terragrunt configuration files (syntactic changes vs semantic changes). This means that
+you won't be making changes to the underlying behavior of the Terraform code or how Terraform is called. As such, there
+will be no need for rolling out the changes with `apply`, or performing any state migrations to avoid backward
+incompatibilities.
```

Comment:
Does it make sense to mention that this DRYing up process is purely a code refactor with no changes to infra or state, closer to the top of the guide?

Reply:
Makes sense: addressed in 80f3f5a

Reply:
Cool, lgtm!